unblob/handlers/archive/zip.py

```diff
@@ -115,13 +116,18 @@ def is_zip64_eocd(end_of_central_directory: Instance):
             or end_of_central_directory.offset_of_cd == 0xFFFFFFFF
         )
 
-    @staticmethod
-    def is_zip64_cd_file(file_header: Instance):
+    def is_zip64_cd_file(self, file, start_offset, offset):
         # see https://pkware.cachefly.net/webdocs/casestudies/APPNOTE.TXT section 4.3.9.2
-        return (
+        file_header = self.cparser_le.partial_cd_file_header_t(file)
+        has_zip64_tag = (
             file_header.file_size == 0xFFFFFFFF
             or file_header.compress_size == 0xFFFFFFFF
         )
+        if has_zip64_tag:
+            with contextlib.suppress(InvalidInputFormat):
+                self._parse_zip64(file, start_offset, offset)
```

Comment:
We need to parse the zip64 structure to decide if it is a zip64, but if it is, we will parse again later:
https://github.com/onekey-sec/unblob/pull/901/files#diff-350302c56e7fd3d4ae27392f6c700612f4ed86cd04cea098f32c5cc67894d30fR182-R184

This work should be possible to spare by some reorganization.